core/async/src/lib.rs

```diff
@@ -132,16 +141,22 @@ impl ActorSystem {
     ///
     /// This is useful for keeping track of spawned futures and their lifetimes.
     /// Behind the scenes, this builds a new EmptyActor each time.
-    pub fn new_future_spawner(&self) -> Box<dyn FutureSpawner> {
-        let handle = self.spawn_tokio_actor(EmptyActor);
+    pub fn new_future_spawner(&self, description: &str) -> Box<dyn FutureSpawner> {
+        let handle = spawn_tokio_actor(
+            EmptyActor,
+            description.to_string(),
+            self.tokio_cancellation_signal.clone(),
+        );
         handle.future_spawner()
     }
 }
 
 /// Spawns a future spawner that is NOT owned by any ActorSystem.
 /// Rather, the returned FutureSpawner, when dropped, will stop the runtime.
-pub fn new_owned_future_spawner() -> Box<dyn FutureSpawner> {
-    Box::new(OwnedFutureSpawner { handle: spawn_tokio_actor(EmptyActor, CancellationToken::new()) })
+pub fn new_owned_future_spawner(description: &str) -> Box<dyn FutureSpawner> {
+    Box::new(OwnedFutureSpawner {
+        handle: spawn_tokio_actor(EmptyActor, description.to_string(), CancellationToken::new()),
```

Comment:
Okay, I see the issue and why we have `actor_name` as a separate field here. Might I suggest the following... We can add a new function to the actor trait called `description()` that has a default implementation that returns `pretty_type_name()`, but we can override the function for EmptyActor to return the `description`.